internal/test/e2e/cloudwatch.go

```diff
@@ -0,0 +1,85 @@
+package e2e
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+var svc *cloudwatch.CloudWatch
+
+func init() {
+	if s, err := session.NewSession(); err == nil {
+		svc = cloudwatch.New(s)
+	} else {
+		fmt.Println("Cannot create CloudWatch service", err)
+	}
+}
+
+func putInstanceTestResultMetrics(r instanceTestsResults) {
+	if svc == nil {
+		logger.Info("Cannot publish metrics as cloudwatch service was not initialized")
+		return
+	}
+
+	logger.Info("Publishing instance test result metrics")
+	erroredCount, failedCount, succeededCount := 0, 0, 0
+	if r.err != nil {
+		erroredCount = 1
+	} else if !r.testCommandResult.Successful() {
+		failedCount = 1
+	} else {
+		succeededCount = 1
+	}
+
+	data := &cloudwatch.MetricDatum{
+		Unit: aws.String("Count"),
+		Dimensions: []*cloudwatch.Dimension{
+			{
+				Name:  aws.String("Provider"),
+				Value: aws.String(getProviderName(r.conf.Regex)),
+			},
+			{
+				Name:  aws.String("BranchName"),
+				Value: aws.String(r.conf.BranchName),
+			},
+		},
+		Timestamp: aws.Time(time.Now()),
+	}
+	putMetric(data, "ErroredInstanceTests", erroredCount)
+	putMetric(data, "FailedInstanceTests", failedCount)
+	putMetric(data, "SucceededInstanceTests", succeededCount)
+
+	// TODO: publish time metrics
+	logger.Info("Test instance metrics published")
+}
+
+func getProviderName(testRe string) string {
+	providerRe := regexp.MustCompile(`Test([A-Z].*?)[A-Z]`)
```

Comment:
Does this regex work for the vsphere test names? For example `TestVSphereKubernetes128To129StackedEtcdRedHatUpgrade`

Reply:
it doesn't, good catch. I have updated the regexp with verification: https://go.dev/play/p/Ry5fK_qW64F

Reply:
v2: https://go.dev/play/p/XaYkT1zWERQ